### PR TITLE
Remove old api usage in cardano-node

### PR DESCRIPTION
**Pull request by Jimbo4350**
- Add BlockIssuer Key instance to the typed api

### CODE REVIEW

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 1]
```diff
@@ -2853,6 +2855,75 @@ instance HasTextEnvelope (SigningKey StakePoolKey) where
...
+    deserialiseFromRawBytes (AsHash AsBlockIssuerKey) bs =
+      BlockIssuerKeyHash . Shelley.KeyHash <$> Crypto.hashFromBytes bs
+
+instance HasTextEnvelope (VerificationKey BlockIssuerKey) where
+    textEnvelopeType _ = "Node operator block issuer verification key"
+    -- TODO: include the actual crypto algorithm name, to catch changes
+
+instance HasTextEnvelope (SigningKey BlockIssuerKey) where
+    textEnvelopeType _ = "Node operator block issuer signing key"
+    -- TODO: include the actual crypto algorithm name, to catch changes
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api/Typed.hs:2925]: Hmm. I don't think this is going to work.

The problem is we want to be able to read in either a genesis key delegate or a stake pool signing key, and then cast them to a block issuer to make the op cert. But those have different text envelope types right?

In fact I don't think we will actually need to serialise a block issuer key role at all.

Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/Typed.hs:2925]: :+1: I will remove `BlockIssuerKey` just use `Shelley.coerceKeyRole` where appropriate

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 2]
```diff
@@ -54,6 +54,10 @@ module Cardano.Api.Typed (
     Crypto.mkSeedFromBytes,
     Crypto.readSeedFromSystemEntropy,
 
+    -- ** Converting keys
+    Shelley.coerceKeyRole,
```
[the remark this answers is not in the document]
Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/Typed.hs:58]: We decided against implementing `BlockIssuerKey` in the typed api but we need to convert from a stake pool verification key to a block issuer key: https://github.com/input-output-hk/cardano-node/pull/1384/files#diff-aa9af82ceb8bf43b76fc52caa281a558R150

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**